twilight-model/src/guild/onboarding/option.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{
+    guild::Emoji,
+    id::{
+        marker::{ChannelMarker, EmojiMarker, OnboardingPromptOptionMarker, RoleMarker},
+        Id,
+    },
+};
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+
+/// An emoji for a guild onboarding prompt.
+/// This is used instead [`Emoji`] as both it's id and name can be `null` in prompt options.
+///
+/// [`Emoji`]: crate::guild::Emoji
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct OnboardingPromptEmoji {
+    name: Option<String>,
+    id: Option<Id<EmojiMarker>>,
+    #[serde(default)]
+    animated: bool,
+}
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum FromOnboardingPromptEmojiError {
+    MissingId,
+    MissingName,
+}
+
+impl Display for FromOnboardingPromptEmojiError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::MissingId => f.write_str("missing emoji id"),
+            Self::MissingName => f.write_str("missing emoji name"),
+        }
+    }
+}
+
+impl TryFrom<OnboardingPromptEmoji> for Emoji {
+    type Error = FromOnboardingPromptEmojiError;
+
+    fn try_from(value: OnboardingPromptEmoji) -> Result<Self, Self::Error> {
+        let OnboardingPromptEmoji { name, id, animated } = value;
+
+        let name = name.ok_or(FromOnboardingPromptEmojiError::MissingName)?;
+        let id = id.ok_or(FromOnboardingPromptEmojiError::MissingId)?;
+
+        Ok(Self {
+            animated,
+            available: bool::default(),
+            id,
+            name,
+            managed: bool::default(),
+            require_colons: bool::default(),
+            roles: Vec::default(),
+            user: Option::default(),
+        })
+    }
+}
+
+impl From<Emoji> for OnboardingPromptEmoji {
+    fn from(value: Emoji) -> Self {
+        Self {
+            animated: value.animated,
+            id: Some(value.id),
+            name: Some(value.name),
+        }
+    }
+}
+
+/// A prompt option for a guild onboarding screen.
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct OnboardingPromptOption {
```

Comment:
Remove initial "The" and "A" prefixes, `s/id/ID`, and end each sentence with a dot.